### DISCUSSION TITLE
Install to a path with space in it

### DISCUSSION THREAD

**Discussion by z0000l** [chocolatey/choco#2448] [Q&A]
Hello everyone, I am having a hard time trying to install the packages to the specific path. I went through the official docs and found a useful command like

> choco install git -y --install-arguments="'/DIR=C:\git'"

And I actually want to install it to my "Program Files" folder in my D drive, so I try something like

> choco install git -y --install-arguments="'/DIR=D:\Program Files'"

and

> choco install git -y --install-arguments="'/DIR=D:\\"Program Files"'"

or run a powershell script like

> $path = "D:\Program Files\"
> choco install lightshot -y --install-arguments="'/DIR=$path'"

and of course none of them worked, they either just get errors and stop or create a new folder called "Program" and put the stuffs in it.

Sorry for asking such stupid questions, I am a newbie to powershell and choco. I will be really grateful if you guys can tell me the right way to do it.

**Comment by pauby** [comment]
Can you try:

```
choco install git -y --install-arguments="'/DIR=""D:\Program Files""'"
```

This is a 2 x double quotes BEFORE `D:\` and a 2 x double quotes after `Files` and then a a single quote, then a double quote.

**Reply by z0000l** [reply]
> 
> 
> Can you try:
> 
> ```
> choco install git -y --install-arguments="'/DIR=""D:\Program Files""'"
> ```
> 
> This is a 2 x double quotes BEFORE `D:\` and a 2 x double quotes after `Files` and then a a single quote, then a double quote.

Doesn't work, still got installed in "Program".

**Reply by pauby** [reply]
Can you try running it with the `-dv` switch, check the Chocolatey logs and see what the command the installer is executing?

**Reply by z0000l** [reply]
> 
> 
> Can you try running it with the `-dv` switch, check the Chocolatey logs and see what the command the installer is executing?

Okay, so I tried installing something else, these are the command history, I abridged the irrelevant:

```
PS C:\WINDOWS\system32> choco install lightshot -y -dv --install-arguments="'/DIR=""D:\Program Files""'"
...
Command line: "C:\ProgramData\chocolatey\choco.exe" install lightshot -y -dv --install-arguments='/DIR="D:\Program Files"'
...
Received arguments: install lightshot -y -dv --install-arguments='/DIR=D:\Program Files'
...
InstallArguments='/DIR=D:\Program Files'|OverrideArguments='False'|
...
Calling built-in PowerShell host with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\lightshot.install\tools\chocolateyinstall.ps1' -installArguments '/DIR=D:\Program Files' -packageParameters ''']
...
Running 'ChocolateyScriptRunner' for lightshot.install v5.5.0.7 with packageScript 'C:\ProgramData\chocolatey\lib\lightshot.install\tools\chocolateyinstall.ps1', packageFolder:'C:\ProgramData\chocolatey\lib\lightshot.install', installArguments: '/DIR=D:\Program Files', packageParameters: '',
...
Running Start-ChocolateyProcessAsAdmin -validExitCodes '0 3010 1641' -workingDirectory 'C:\Users\shich\AppData\Local\Temp\chocolatey\lightshot.install\5.5.0.7' -statements '/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP- /DIR=D:\Program Files' -exeToRun 'C:\Users\shich\AppData\Local\Temp\chocolatey\lightshot.install\5.5.0.7\setup-lightshot.exe'
...
Elevating permissions and running ["C:\Users\shich\AppData\Local\Temp\chocolatey\lightshot.install\5.5.0.7\setup-lightshot.exe" /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP- /DIR=D:\Program Files]. This may take a while, depending on the statements.
Command ["C:\Users\shich\AppData\Local\Temp\chocolatey\lightshot.install\5.5.0.7\setup-lightshot.exe" /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP- /DIR=D:\Program Files] exited with '0'.
...
Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\lightshot.install\tools\chocolateyinstall.ps1' -installArguments '/DIR=D:\Program Files' -packageParameters '''] exited with '0'.
...
Software installed to 'D:\Program\'
```

So it seems like it has been sticking to the correct install argument but it suddenly turned around at the last moment.
